File: app-android/app/src/main/java/de/tutao/tutanota/alarms/RepeatPeriod.java
package de.tutao.tutanota.alarms;

public enum RepeatPeriod {
	DAILY, WEEKLY, MONTHLY, ANNUALLY;

	public static RepeatPeriod get(long value) {
		if (value > RepeatPeriod.values().length - 1) {
			throw new IllegalArgumentException("Unknown RepeatPeriod: " + value);
		}
		return RepeatPeriod.values()[(int) value];
	}
}

File: app-android/tutashared/src/main/java/de/tutao/tutashared/alarms/RepeatPeriod.kt
package de.tutao.tutashared.alarms

enum class RepeatPeriod {
	DAILY,
	WEEKLY,
	MONTHLY,
	ANNUALLY;

	fun value(): Long {
		return ordinal.toLong()
	}

	companion object {
		operator fun get(value: Long): RepeatPeriod {
			require(value <= values().size - 1) { "Unknown RepeatPeriod: $value" }
			return values()[value.toInt()]
		}
	}
}